apps/prairielearn/src/tests/questionSharing.test.ts

```diff
@@ -0,0 +1,252 @@
+        }).toString(),
+      });
+
+      response = await helperClient.fetchCheerio(sharingUrl);
+      const result = UUID_REGEXP.exec(response.text());
+      exampleCourseSharingId = result ? result[0] : null;
+      assert(exampleCourseSharingId != null);
+    });
+
+    step('Create a sharing set', async () => {
+      const sharingUrl = sharingPageUrl(testCourseId);
+      const response = await helperClient.fetchCheerio(sharingUrl);
+      const token = response.$('#test_csrf_token').text();
+      await fetch(sharingUrl, {
+        method: 'POST',
+        headers: { 'Content-type': 'application/x-www-form-urlencoded' },
+        body: new URLSearchParams({
+          __action: 'unsafe_sharing_set_create',
+          __csrf_token: token,
+          sharing_set_name: sharingSetName,
+        }).toString(),
+      });
+
+      const sharingPage = await (await fetch(sharingPageUrl(exampleCourseId))).text();
+      assert(sharingPage.includes(exampleCourseSharingName));
+    });
+
+    // step('Attempt to create another sharing set with the same name', async () => {
+    //   // TODO ensure that the sharing set name you created only appears once on the page
+    // });
+
+    // step('Attempt to create a sharing set with an invalid name', async () => {
+
+    // });
+
+    step('Share sharing set with example course', async () => {
+      const sharingUrl = sharingPageUrl(testCourseId);
+      const response = await helperClient.fetchCheerio(sharingUrl);
+      const token = response.$('#test_csrf_token').text();
+      await fetch(sharingUrl, {
+        method: 'POST',
+        headers: { 'Content-type': 'application/x-www-form-urlencoded' },
+        body: new URLSearchParams({
+          __action: 'unsafe_course_sharing_set_add',
+          __csrf_token: token,
+          sharing_set_id: '1',
+          course_sharing_id: exampleCourseSharingId,
+        }).toString(),
+      });
+
+      const sharingPage = await (await fetch(sharingPageUrl(testCourseId))).text();
+      assert(sharingPage.includes('XC 101'));
+    });
+
+    // step('Attempt to share sharing set with invalid course ID', async () => {
+
+    // });
+
+    // step('Attempt to create another sharing set with the same name', async () => {
+
+    // });
+
+    step('Add question "addNumbers" to sharing set', async () => {
+      // TODO: should this block of code be factored out to helperClient as a
+      // helper function for getting to the page of a question with a given qid?
+      // or does this code already exist somewhere and I am duplicating effort here?
+      const questionsUrl = `${baseUrl}/course/${testCourseId}/course_admin/questions`;
+      const questionsPage = await helperClient.fetchCheerio(questionsUrl);
+      const questionData = questionsPage.$('#questionsTable').attr('data-data');
+      const questions = JSON.parse(questionData);
+      const addNumbersInfo = questions.find((questionInfo) => questionInfo.qid === 'addNumbers');
```

Comment:
Feel free to just hit the database directly for this; it'll be marginally faster.

Reply:
sure, done.